site/src/main/java/uk/nhs/digital/ps/beans/BaseDocument.java

```diff
@@ -63,4 +65,29 @@ protected RestrictableDate nominalPublicationDateCalendarToRestrictedDate(final
             ? RestrictableDate.restrictedDateFrom(nominalPublicationDate)
             : RestrictableDate.fullDateFrom(nominalPublicationDate);
     }
+
+    /*
+    * Converts selected geographic coverage values:
+    * If England, Wales, Scotland are checked and no others - Great Britain
+    * If England, Wales, Scotland and Northern Ireland are checked and no others - United Kingdom
+    * If England, Wales, Scotland and Northern Ireland, and Republic of Ireland are ticked - British Isles
+    * If any other combination is checked, serve up a list of the checked values
+    */
+    protected String[] geographicCoverageValuesToRegionValue(final String[] cmsValues) {
+
+        if (cmsValues == null) {
+            return null;
+        }
+
+        Set<String> selectedCmsValues = new HashSet<>(Arrays.asList(cmsValues));
+        if (GREAT_BRITAIN.equals(selectedCmsValues)) {
```

Comment:
It would be good to introduce an enum called something like 'Region' and move this logic there; `BaseDocument` has no business doing this resolution.